include/xtensor/xassign.hpp

```diff
@@ -289,12 +291,27 @@ namespace xt
     {
         using shape_type = typename E1::shape_type;
         using size_type = typename E1::size_type;
+        using S1 = typename E1::shape_type;
+        using S2 = typename E2::shape_type;
```

Comment:
we already have `shape_type = typename E1::shape_type`, that would be better to reuse it instead of defining twice the same type (maybe rename the type shape_type1 and shape_type2)

Reply:
:+1: